kimchi/src/prover.rs

```diff
@@ -633,7 +635,7 @@ where
 
             if let Some(selector) = index.cs.foreign_field_add_selector_poly.as_ref() {
                 index_evals.extend(
-                    foreign_field_add::gadget::circuit_gates()
+                    [GateType::ForeignFieldAdd]
```

Comment:
Replace with a `push` (although really, this PR shouldn't be touching any other gates at all..)

Reply:
I guess I started modifying these because I received some comments suggesting to remove these methods when gadgets had one single gate type (as opposed to rangecheck), and then included the changes in the same branch.